### ISSUE TITLE
[docs] Best Practices and Guidelines ("This / Not This") examples improvements

### ISSUE THREAD

**Issue by markcaron** [RedHat-UX/red-hat-design-system#1449]
Break up our "This / Not This" best practices and guidelines images and examples into separate images and columns (or cards), similar to what we do on our Brand site.

This will make it easier to scan as well as improve non-visual accessibility.

"This" and "Not this" should be HTML text paired with an icon.

### Current situation
<img width="1065" alt="image" src="https://github.com/RedHat-UX/red-hat-design-system/assets/1890950/f3776e75-9fda-4b8d-9c7a-381cde55e812">

### What we're looking for
<img width="883" alt="image" src="https://github.com/RedHat-UX/red-hat-design-system/assets/1890950/13365a65-a794-4524-97b3-609d7bf6d870">

**Comment by marionnegp** [comment]
The Color page's [Best Practices](https://ux.redhat.com/foundations/color/usage/#best-practices) section uses this format.

This is a sample of what the Dos and Don'ts currently look like in code:

```
<div class="best-practices-grid">
    <div>
        <img slot="header" src="/assets/color/best-practices-custom-themes-correct.svg" alt="Blue submit button, blue default call to action, black tooltip, and blue switch">
        <h4 class="correct">Do</h4>
        <p>Use the color variants already available for elements and patterns.</p>
    </div>
    <div>
        <img slot="header" src="/assets/color/best-practices-custom-themes-wrong.svg" alt="Magenta button, brand red default call to action, green tooltip, and dark orange switch">
        <h4 class="wrong">Don't</h4>
        <p>Do not create your own colors, design tokens, or change the colors of existing elements and patterns.</p>
    </div>
</div>
```

It would be good to add this format as a shortcode.

**Comment by marionnegp** [comment]
Also, is there a preference for using "This"/"Not this" vs "Do"/"Don't"?

**Comment by markcaron** [comment]
> Also, is there a preference for using "This"/"Not this" vs "Do"/"Don't"?

@marionnegp not really. I'm ok with using Do/Don't instead. Might be more of a common practice anyway.

**Comment by zeroedin** [comment]
I did some initial work on this in #1465 

https://deploy-preview-1465--red-hat-design-system.netlify.app/foundations/color/usage/#custom-themes

The variation looks a bit different then the examples above as I used a `<figure>` For the image.  It seemed appropriate given the linked description of the images as the `<figcaption>`

![image](https://github.com/RedHat-UX/red-hat-design-system/assets/13666/c6553932-e14e-408e-ba72-081e8578f1b7)

Here is what the HTML looks like:

```
<uxdot-best-practice>
  <uxdot-example slot="do" no-border="" variant="full" alignment="left" color-palette="lightest">
    <figure>
      <img slot="header" src="/assets/color/best-practices-custom-themes-correct.svg" alt="Blue submit button, blue default call to action, black tooltip, and blue switch">
      <figcaption>Use the color variants already available for elements and patterns.</figcaption>
    </figure>
  </uxdot-example>
  <uxdot-example slot="dont" no-border="" variant="full" alignment="left" color-palette="lightest">
    <figure>
      <img slot="header" src="/assets/color/best-practices-custom-themes-wrong.svg" alt="Magenta button, brand red default call to action, green tooltip, and dark orange switch">
      <figcaption>Do not create your own colors, design tokens, or change the colors of existing elements and patterns.</figcaption>
    </figure>
  </uxdot-example>
</uxdot-best-practice>
```

This component can be tweaked however we deem necessary as I was just evaluating using a slotted `<figure>` in a `uxdot-example` for this use case.  

Right now `<uxdot-best-practice>` has 2 slots `do` and `dont` and currently only only displays in a 2 up grid as that is the only use case I've ran across so far.

**Comment by marionnegp** [comment]
@zeroedin, using `<figcaption>` makes sense to me, though we might want to adjust the placement of the caption or "Do/Don't" heading.

There might be times we want to add more than one "do" or "don't" image (like if we split the two images in this [audio player example](https://ux.redhat.com/elements/audio-player/guidelines/#contextual-menu)) or have a best practice example be full-width.

**Comment by zeroedin** [comment]
`<uxdot-best-practice do>` and `<uxdot-best-practice dont>` which use a figcaption in the shadowroot.  It is available in #1465.  This PR does not update any content.  

I believe this issue partially an ask to update all `Best Practice` sections across the site to use this new style as well so will leave it open.

**Comment by coreyvickery** [comment]
@zeroedin I like the design below if we can change the old icons for new brand icons as well as adding padding in between the icon and text.

How do we move this forward?

<img width="1162" alt="Screenshot 2024-10-26 at 10 29 22 PM" src="https://github.com/user-attachments/assets/f5157104-c6d2-4869-81d8-67c49aa211ee">
